src/pages/academy/gameSimulator/subcomponents/chapterPublisher/ChapterPublisher.tsx

```diff
@@ -0,0 +1,42 @@
+import { memo, useState } from 'react';
+import { useRequest } from 'src/commons/utils/Hooks';
+import { defaultChapter } from 'src/features/gameSimulator/GameSimulatorConstants';
+import { fetchChapters, fetchTextAssets } from 'src/features/gameSimulator/GameSimulatorService';
+import { ChapterDetail } from 'src/features/gameSimulator/GameSimulatorTypes';
+
+import ChapterPublisherEditor from './ChapterPublisherEditor';
+
+/**
+ * This components renders the Chapter Publisher component in the Game Simulator.
+ *
+ * @param textAssets - List of all text assets on S3 to choose from.
+ */
+const ChapterPublisher = memo(() => {
```

Comment:
Nit: Can we, to keep it consistent with the other codebases:

```tsx
import React, { memo, useState } from 'react';
// ...
const ChapterPublisher: React.FC = () => { /* ... */ };

// ...
export default memo(ChapterPublisher, () => true);
```

Also, can I check, should the component never be recreated after it's been created once? If yes, I'd say `() => true` is preferred in the memoization condition.

Reply:
Addressed in [962b341](https://github.com/source-academy/frontend/pull/2836/commits/962b3413e413ff74052dd0e7f90ff5830ec709fb) - and to answer the second part, `() => true` is applicable for this component, yes 👍